neurom/core/_neuron.py

```diff
@@ -291,19 +407,69 @@ def __str__(self):
     __repr__ = __str__
 
 
-class Neuron:
+class Neuron(morphio.mut.Morphology):
     """Class representing a simple neuron."""
 
-    def __init__(self, soma=None, neurites=None, sections=None, name='Neuron'):
-        """Initialize a Neuron object."""
-        self.soma = soma
-        self.name = name
-        self.neurites = neurites
-        self.sections = sections
+    def __init__(self, filename, name=None):
+        """Neuron constructor.
+
+        Args:
+            filename (str|Path): a filename
+            name (str): a option neuron name
+        """
+        try:
+            morphio.set_ignored_warning([morphio.Warning.appending_empty_section,
+                                         morphio.Warning.wrong_root_point], True)
+            morphio.set_raise_warnings(True)
+            super().__init__(filename)
+        finally:
+            morphio.set_raise_warnings(False)
+        self.name = name if name else 'Neuron'
+        self.morphio_soma = super().soma
+        self.neurom_soma = make_soma(self.morphio_soma)
+
+    @property
+    def soma(self):
+        """Corresponding soma."""
+        return self.neurom_soma
+
+    @property
+    def neurites(self):
+        """The list of neurites."""
+        return [Neurite(root_section) for root_section in self.root_sections]
+
+    @property
+    def sections(self):
+        """The array of all sections, excluding the soma."""
+        return list(iter_sections(self))
+
+    @property
+    def points(self):
+        """Returns the list of points."""
+        return np.concatenate(
+            [section.points for section in iter_sections(self)])
+
+    def transform(self, trans):
+        """Return a copy of this neuron with a 3D transformation applied."""
+        obj = Neuron(self)
+        obj.morphio_soma.points = trans(obj.morphio_soma.points)
+
+        for section in obj.sections:
+            section.morphio_section.points = trans(section.morphio_section.points)
+        return obj
+
+    def __copy__(self):
+        """Creates a deep copy of Neuron instance."""
+        return Neuron(self, self.name)
+
+    def __deepcopy__(self, memodict={}):
+        """Creates a deep copy of Neuron instance."""
+        # pylint: disable=dangerous-default-value
+        return Neuron(self, self.name)
 
     def __str__(self):
         """Return a string representation."""
         return 'Neuron <soma: %s, n_neurites: %d>' % \
-            (self.soma, len(self.neurites))
+            ('1', len(self.neurites))
```

Comment:
I didn't pay attention to that. I think we can keep the old solution. I don't know why Benoit made it. Probably because soma now always considered as one section in terms of NeuroM v1.

Reply:
done